tools/c7n_gcp/c7n_gcp/resources/bigtable.py

```diff
@@ -0,0 +1,157 @@
+        return {
+            'parent': '{}'.format(
+                parent_instance['name'],
+            )
+        }
+
+
+@BigTableInstanceClusterBackup.filter_registry.register('time-range')
+class TimeRange(Filter):
+
+    schema = type_schema('time-range',
+                         value={'$ref': '#/definitions/filters_common/value'})
+    permissions = ('bigtable.backups.list',)
+
+    datetime1_pattern = "%Y-%m-%dT%H:%M:%S.%fZ"
+    datetime2_pattern = "%Y-%m-%dT%H:%M:%S"
+
+    def process(self, resources, event=None):
+        filtered_resources = []
+        value = self.data.get('value')
+        field1 = 'startTime'
+        field2 = 'expireTime'
+        for resource in resources:
+            start_time_pattern = self.datetime1_pattern
+            expired_time_pattern = self.datetime1_pattern
+            expired_time = resource[field2]
+            start_time = resource[field1]
+            if '.' not in expired_time and 'Z' in expired_time:
+                expired_time_pattern = self.datetime2_pattern
+                expired_time = expired_time[:-1]
+            if '.' not in start_time and 'Z' in start_time:
+                start_time_pattern = self.datetime2_pattern
+                start_time = start_time[:-1]
+
+            filtered_expired_time = datetime.datetime.strptime(
+                expired_time, expired_time_pattern)
+            filtered_start_time = datetime.datetime.strptime(
+                start_time, start_time_pattern)
+            result_time = filtered_expired_time - filtered_start_time
+            if int(result_time.days) < value:
+                filtered_resources.append(resource)
+
+        return filtered_resources
+
+
+@resources.register('bigtable-instance-table')
+class BigTableInstanceTable(ChildResourceManager):
+    """ GC resource:
+    https://cloud.google.com/bigtable/docs/reference/admin/rest/v2/projects.instances.tables"""
+    class resource_type(ChildTypeInfo):
+        service = 'bigtableadmin'
+        version = 'v2'
+        component = 'projects.instances.tables'
+        enum_spec = ('list', 'tables[]', None)
+        scope = 'parent'
+        name = id = 'name'
+        parent_spec = {
+            'resource': 'bigtable-instance',
+            'child_enum_params': {
+                ('name', 'parent')},
+            'use_child_query': True,
+        }
+        default_report_fields = ['name']
+        permissions = ('bigtable.tables.list',)
+
+    def _get_child_enum_args(self, parent_instance):
+        return {
+            'parent': 'projects/{}/instances/{}'.format(
+                local_session(self.session_factory).get_default_project(),
+                parent_instance['displayName'],
+            )
+        }
+
+
+@BigTableInstanceTable.filter_registry.register('iam-policy')
+class BigTableInstanceTableIamPolicyFilter(IamPolicyFilter):
+    """
+    Overrides the base implementation to process service account resources correctly.
```

Comment:
filter/action doc strings are for end user reference docs on usage, not contributors. please adjust verbiage here, or remove for the base class doc string to pass through.

Reply:
fixed